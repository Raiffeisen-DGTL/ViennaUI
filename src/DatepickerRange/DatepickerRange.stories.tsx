import React from 'react';
import { Story, Meta } from 'storybook';
import { DatepickerRange, DatepickerRangeProps } from './DatepickerRange';

export default {
    title: 'Development/DatepickerRange',
    component: DatepickerRange,
    argTypes: {
        value: {
            control: false,
        },
    },
} as Meta;

export const Overview: Story<DatepickerRangeProps> = () => {
    const [value, setValue] = React.useState();
    const handleChange = React.useCallback((e, data) => {
        setValue(data.value);
    }, []);
    return <DatepickerRange value={value} onChange={handleChange} />;
};
