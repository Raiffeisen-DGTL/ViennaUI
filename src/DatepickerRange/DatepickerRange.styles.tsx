import styled, { css } from 'styled-components';
import { getPresets } from '../Utils/styling';

const presets = getPresets('datepickerrange', {
    calendar: null,
});

export const Box = styled.div`
    position: relative;
    display: flex;
    flex-direction: column;
`;

export const InputBox = styled.div`
    position: relative;
`;

interface CalendarBoxProps {
    $hidden: boolean;
}
export const CalendarBox = styled.div<CalendarBoxProps>`
    ${presets.calendar};
    ${({ $hidden }) =>
        $hidden &&
        css`
            display: none;
        `};
`;
