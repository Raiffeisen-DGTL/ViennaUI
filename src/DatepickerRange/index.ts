export { DatepickerRange, type DatepickerRangeProps } from './DatepickerRange';
export type { DatePickerRangeLocalization } from './localization';
