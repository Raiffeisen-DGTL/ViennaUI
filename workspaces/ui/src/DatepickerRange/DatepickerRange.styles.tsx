import styled from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const presets = getPresets('datepickerrange', {
    calendar: null,
});

export const Box = styled.div`
    position: relative;
    display: flex;
    flex-direction: column;
`;

export const InputBox = styled.div`
    position: relative;
`;

export const CalendarBox = styled.div`
    position: absolute;
    opacity: 0;
    overflow: hidden;
    ${presets.calendar};
`;
