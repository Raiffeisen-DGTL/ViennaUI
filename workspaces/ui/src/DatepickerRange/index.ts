export * from './DatepickerRange';
export type { DatePickerRangeLocalization } from './localization';
