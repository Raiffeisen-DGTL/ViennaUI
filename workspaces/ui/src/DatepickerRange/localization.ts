import { DSLocalizationProps } from '../Localization';
import { CalendarLocalization } from '../Calendar';
import { InputDateRangeLocalization } from '../InputMask';

export type DatePickerRangeLocalization = CalendarLocalization & InputDateRangeLocalization;

export type DatePickerRangeLocalizationProps = DSLocalizationProps<DatePickerRangeLocalization>;
