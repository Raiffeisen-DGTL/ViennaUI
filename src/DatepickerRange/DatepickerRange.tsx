import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import { Calendar as CalendarIcon } from 'vienna.icons';
import { Locale } from 'date-fns';
import { useFloating, offset, autoPlacement, FloatingPortal } from '@floating-ui/react';
import { CalendarProps } from 'vienna.icons/dist/Calendar/Calendar';
import { usePortal } from 'vienna.react-use';
import { DatePickerRangeLocalizationProps } from './localization';
import { Calendar, dateFunction, Dates, DateValue } from '../Calendar';
import { InputDateRange, DateType } from '../InputMask';
import { getDateFromString, getStringFromDate } from '../Utils/DateUtils';
import { InputEvent, InputProps } from '../Input';
import { Box, InputBox, CalendarBox } from './DatepickerRange.styles';
import { checkIsDisabled } from '../Calendar/Utils';
import { StartingWeekDay } from '../Calendar/types';

export interface DatepickerRangeProps
    extends Omit<InputProps, 'onChange' | 'onPaste'>,
        DatePickerRangeLocalizationProps {
    /**
     * дата передается строкой вида 01.01.2021-12.01.2021
     */
    value?: string;

    /**
     * Отображение календаря
     */
    isCalendarOpen?: boolean;

    /**
     * Даты-событие
     */
    eventDates?: Date[] | dateFunction;

    /**
     * Нижняя граница выбора даты
     */
    minDate?: Date;

    /**
     * Верхняя граница выбора даты
     */
    maxDate?: Date;

    onChange?: (
        event: InputEvent<React.FormEvent<HTMLInputElement>> | Event | null,
        data: { value?: string; valueAsDate: DateType; name?: string; isDisabled?: boolean }
    ) => void;

    /**
     * Неактивные для выбора даты
     */
    disabledDates?: Dates;

    /**
     * Дни недели отображаются с понедельника (1) или с воскресенья (0)
     */
    startingWeekDay?: StartingWeekDay;
    /**
     * Локаль календаря
     */
    locale?: Locale;
    /**
     * Изменение отображаемого месяца/года
     */
    onChangeDisplayedDate?: (date: Date) => void;
    /**
     * Дефолтное значение отображаемой даты
     */
    defaultDisplayedDate?: Date;
}

export const DatepickerRange: React.FC<DatepickerRangeProps> = ({
    eventDates,
    isCalendarOpen = false,
    maxDate,
    minDate,
    name,
    postfix,
    prefix,
    size,
    value,
    onChange,
    onFocus,
    onKeyPress,
    disabledDates,
    startingWeekDay = 'monday',
    onBlur,
    tabIndex,
    localization,
    locale,
    onChangeDisplayedDate,
    defaultDisplayedDate,
    ...attrs
}) => {
    const [isOpen, setOpen] = useState<boolean | undefined>(isCalendarOpen);
    const [active, setActive] = useState(false);
    const datepickerRangeEl = useRef<HTMLDivElement>(null);
    const [isDisabled, setIsDisabled] = useState<boolean>(false);
    const containerPortal = usePortal('box', 'datepickerrange-box');

    const { refs, floatingStyles, update } = useFloating({
        placement: 'bottom-start',
        middleware: [
            offset({
                mainAxis: 4,
                crossAxis: 0,
                alignmentAxis: 0,
            }),
            autoPlacement({
                alignment: 'start',
                allowedPlacements: ['top-start', 'bottom-start'],
                crossAxis: false,
            }),
        ],
    });

    useEffect(() => {
        if (isOpen) {
            update();
        }
    }, [isOpen]);

    useEffect(() => {
        setOpen(isCalendarOpen);
    }, [isCalendarOpen]);

    const handleInputFocus = useCallback(
        (event, data) => {
            if (typeof onFocus === 'function') {
                onFocus(event, { value: data.value, name });
            }
            setOpen(true);
        },
        [name, onFocus]
    );

    const handleClickInput = useCallback(() => {
        if (active) {
            setOpen(true);
        }
    }, [isOpen, active]);

    const getInputValueArray = useCallback((dateString) => {
        return typeof dateString === 'string' ? dateString.replace(/\s+/g, '').split('-') : [];
    }, []);

    const handleChangeInput = useCallback(
        (value) => {
            const inputValueArray = getInputValueArray(value);
            const startInputDate = getDateFromString(inputValueArray[0]);
            const endInputDate = getDateFromString(inputValueArray[1]);

            const isDisabled =
                (!!startInputDate &&
                    checkIsDisabled({ dates: disabledDates, date: startInputDate, startingWeekDay })) ||
                (!!endInputDate && checkIsDisabled({ dates: disabledDates, date: endInputDate, startingWeekDay }));
            setIsDisabled(isDisabled);

            if (typeof onChange === 'function') {
                onChange(null, {
                    value,
                    valueAsDate: { from: startInputDate, to: endInputDate },
                    name,
                    isDisabled,
                });
            }
        },
        [name, onChange]
    );

    const handleClickDate = useCallback(
        (event, { dateStart, dateEnd }: { dateStart?: DateValue | Date; dateEnd?: DateValue | Date }) => {
            let newInputValue = '';
            if (dateStart) {
                newInputValue = getStringFromDate(dateStart as Date);
            }
            if (dateEnd) {
                const formattedStartDate = getStringFromDate(dateStart as Date);

                dateEnd = new Date(dateEnd as Date);
                dateEnd = new Date(dateEnd.setHours(dateEnd.getHours() + 23));
                dateEnd = new Date(dateEnd.setMinutes(dateEnd.getMinutes() + 59));
                dateEnd = new Date(dateEnd.setSeconds(dateEnd.getSeconds() + 59));

                const formattedEndDate = getStringFromDate(dateEnd);

                newInputValue = `${formattedStartDate} - ${formattedEndDate}`;
                isOpen && setOpen(false);
                setActive(true);
            }

            if (typeof onChange === 'function') {
                onChange(event, { value: newInputValue, valueAsDate: { from: dateStart, to: dateEnd }, name });
            }
        },
        [name, onChange, isOpen]
    );

    const handleKeyPress = useCallback(
        (event) => {
            if (typeof onKeyPress === 'function') {
                onKeyPress(event);
            }
            if (event.key === 'Enter') {
                setOpen(!isOpen);
            }
        },
        [isOpen, onKeyPress]
    );

    const handleClickDocument = useCallback((event): void => {
        if (
            event.target instanceof Node &&
            datepickerRangeEl &&
            datepickerRangeEl.current &&
            !datepickerRangeEl.current.contains(event.target)
        ) {
            setOpen(false);
            setActive(false);
        }
    }, []);

    useEffect(() => {
        if (document.body && isOpen) {
            document.body.addEventListener('click', handleClickDocument);
        }

        return () => {
            if (document.body) {
                document.body.removeEventListener('click', handleClickDocument);
            }
        };
    }, [isOpen, handleClickDocument]);

    const dateValue = useMemo(() => {
        const inputValueArray = getInputValueArray(value);
        dateValue?.end?.setSeconds(dateValue?.end?.getSeconds() + 3600 * 23 + 59 * 60 + 59);

        return isDisabled
            ? {}
            : {
                  start: getDateFromString(inputValueArray[0]),
                  end: getDateFromString(inputValueArray[1]),
              };
    }, [value]);

    const handleMouseDownCalendar = useCallback((e) => {
        e.preventDefault();
    }, []);

    const handleBlur = useCallback(
        (event) => {
            if (typeof onBlur === 'function') {
                onBlur(event, {
                    value: `${dateValue.start}-${dateValue.end}`,
                    name,
                });
            }
            setOpen(false);
            setActive(false);
        },
        [onBlur, dateValue, minDate, name]
    );
    return (
        <Box
            ref={refs.setReference}
            onClick={handleClickInput}
            tabIndex={tabIndex}
            onBlur={handleBlur}
            id='datepickerrange-box'>
            <InputBox>
                <InputDateRange
                    {...(attrs as {})}
                    postfix={
                        postfix ?? (
                            <CalendarIcon
                                size={size === 'xxl' ? 'xl' : (size as CalendarProps['size'])}
                                cursor='pointer'
                            />
                        )
                    }
                    prefix={prefix ?? null}
                    size={size}
                    name={name}
                    active={active}
                    value={value}
                    localization={localization}
                    onChange={handleChangeInput}
                    onFocus={handleInputFocus}
                    onKeyPress={handleKeyPress}
                />
            </InputBox>
            <FloatingPortal root={containerPortal}>
                <CalendarBox
                    data-id='calendar'
                    ref={refs.setFloating}
                    style={floatingStyles}
                    $hidden={!isOpen}
                    onMouseDown={handleMouseDownCalendar}>
                    <Calendar
                        dateStart={dateValue.start}
                        dateEnd={dateValue.end}
                        eventDates={eventDates}
                        maxDate={maxDate}
                        minDate={minDate}
                        todayButton={false}
                        ranged
                        disabledDates={disabledDates}
                        startingWeekDay={startingWeekDay}
                        localization={localization}
                        locale={locale}
                        defaultDisplayedDate={defaultDisplayedDate}
                        onChange={handleClickDate}
                        onChangeDisplayedDate={onChangeDisplayedDate}
                    />
                </CalendarBox>
            </FloatingPortal>
        </Box>
    );
};

DatepickerRange.displayName = 'DatepickerRange';
