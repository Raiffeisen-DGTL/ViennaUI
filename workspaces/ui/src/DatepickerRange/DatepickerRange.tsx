import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
import { useDrop } from 'vienna.react-use';
import { Calendar as CalendarIcon } from 'vienna.icons';
import { Locale } from 'date-fns';
import { DatePickerRangeLocalizationProps } from './localization';
import { Calendar, dateFunction, DateValue, DisabledDates } from '../Calendar';
import { InputDateRange, DateType } from '../InputMask';
import { getDateFromString, getStringFromDate } from '../Utils/DateUtils';
import { InputEvent, InputProps } from '../Input';
import { Box, InputBox, CalendarBox } from './DatepickerRange.styles';
import { checkIsDisabled } from '../Calendar/Utils';

export interface DatepickerRangeProps
    extends Omit<InputProps, 'onChange' | 'onPaste'>,
        DatePickerRangeLocalizationProps {
    /**
     * дата передается строкой вида 01.01.2021-12.01.2021
     */
    value?: string;

    /**
     * Отображение календаря
     */
    isCalendarOpen?: boolean;

    /**
     * Даты-событие
     */
    eventDates?: Date[] | dateFunction;

    /**
     * Нижняя граница выбора даты
     */
    minDate?: Date;

    /**
     * Верхняя граница выбора даты
     */
    maxDate?: Date;

    onChange?: (
        event: InputEvent<React.FormEvent<HTMLInputElement>> | Event | null,
        data: { value?: string; valueAsDate: DateType; name?: string; isDisabled?: boolean }
    ) => void;

    /**
     * Неактивные для выбора даты
     */
    disabledDates?: DisabledDates.WEEKENDS | Date[] | dateFunction;

    /**
     * Дни недели отображаются с понедельника (1) или с воскресенья (0)
     */
    startingWeekDay?: 0 | 1;
    /**
     * Локаль календаря
     */
    locale?: Locale;
}

export const DatepickerRange: React.FC<DatepickerRangeProps> = ({
    eventDates,
    isCalendarOpen,
    maxDate,
    minDate,
    name,
    postfix,
    prefix,
    size,
    value,
    onChange,
    onFocus,
    onKeyPress,
    disabledDates,
    startingWeekDay = 1,
    onBlur,
    tabIndex,
    localization,
    locale,
    ...attrs
}: DatepickerRangeProps): JSX.Element => {
    const [isOpen, setOpen] = useState<boolean | undefined>(isCalendarOpen);
    const datepickerRangeEl = useRef<HTMLDivElement>(null);
    const [isDisabled, setIsDisabled] = useState<boolean>(false);
    const calendarRef = useDrop({ align: 'vertical', float: 'start', margins: { x: 0, y: 4 } });

    useEffect(() => {
        setOpen(isCalendarOpen);
    }, [isCalendarOpen]);

    const handleInputFocus = useCallback(
        (event, data) => {
            if (typeof onFocus === 'function') {
                onFocus(event, { value: data.value, name });
            }
            setOpen(true);
        },
        [name, onFocus]
    );

    const handleClickDateIcon = () => {
        setOpen(!isOpen);
    };

    const getInputValueArray = useCallback((dateString) => {
        return typeof dateString === 'string' ? dateString.replace(/\s+/g, '').split('-') : [];
    }, []);

    const handleChangeInput = useCallback(
        (event, data) => {
            const inputValueArray = getInputValueArray(data.value);
            const startInputDate = getDateFromString(inputValueArray[0]);
            const endInputDate = getDateFromString(inputValueArray[1]);
            const isDisabled =
                (!!startInputDate &&
                    checkIsDisabled({ dates: disabledDates, date: startInputDate, startingWeekDay })) ||
                (!!endInputDate && checkIsDisabled({ dates: disabledDates, date: endInputDate, startingWeekDay }));
            setIsDisabled(isDisabled);

            if (typeof onChange === 'function') {
                onChange(event, { value: data.value, valueAsDate: data.valueAsDate, name, isDisabled });
            }
        },
        [name, onChange]
    );

    const handleClickDate = useCallback(
        (event, { dateStart, dateEnd }: { dateStart?: DateValue | Date; dateEnd?: DateValue | Date }) => {
            let newInputValue = '';
            if (dateStart) {
                newInputValue = getStringFromDate(dateStart as Date);
            }

            if (dateEnd) {
                const formattedStartDate = getStringFromDate(dateStart as Date);
                const formattedEndDate = getStringFromDate(dateEnd as Date);
                newInputValue = `${formattedStartDate} - ${formattedEndDate}`;
                setOpen(false);
            }

            if (typeof onChange === 'function') {
                onChange(event, { value: newInputValue, valueAsDate: { from: dateStart, to: dateEnd }, name });
            }
        },
        [name, onChange]
    );

    const handleKeyPress = useCallback(
        (event) => {
            if (typeof onKeyPress === 'function') {
                onKeyPress(event);
            }
            if (event.key === 'Enter') {
                setOpen(!isOpen);
            }
        },
        [isOpen, onKeyPress]
    );

    const handleClickDocument = useCallback((event): void => {
        if (
            event.target instanceof Node &&
            datepickerRangeEl &&
            datepickerRangeEl.current &&
            !datepickerRangeEl.current.contains(event.target)
        ) {
            setOpen(false);
        }
    }, []);

    useEffect(() => {
        if (document.body && isOpen) {
            document.body.addEventListener('click', handleClickDocument);
        }

        return () => {
            if (document.body) {
                document.body.removeEventListener('click', handleClickDocument);
            }
        };
    }, [isOpen, handleClickDocument]);

    const dateValue = useMemo(() => {
        const inputValueArray = getInputValueArray(value);

        return isDisabled
            ? {}
            : {
                  start: getDateFromString(inputValueArray[0]),
                  end: getDateFromString(inputValueArray[1]),
              };
    }, [value]);

    const handleMouseDownCalendar = useCallback((e) => {
        e.preventDefault();
    }, []);

    const handleBlur = useCallback(
        (event) => {
            if (typeof onBlur === 'function') {
                onBlur(event, {
                    value: `${dateValue.start}-${dateValue.end}`,
                    name,
                });
            }
            setOpen(false);
        },
        [onBlur, dateValue, minDate, name]
    );

    return (
        <Box ref={datepickerRangeEl} tabIndex={tabIndex} onBlur={handleBlur}>
            <InputBox>
                <InputDateRange
                    postfix={
                        postfix ?? <CalendarIcon size={size === 'xxl' ? 'xl' : size} onClick={handleClickDateIcon} />
                    }
                    prefix={prefix ?? null}
                    size={size}
                    name={name}
                    value={value}
                    localization={localization}
                    onChange={handleChangeInput}
                    onFocus={handleInputFocus}
                    onKeyPress={handleKeyPress}
                    {...attrs}
                />
            </InputBox>
            {isOpen && (
                <CalendarBox ref={calendarRef} onMouseDown={handleMouseDownCalendar}>
                    <Calendar
                        dateStart={dateValue.start}
                        dateEnd={dateValue.end}
                        eventDates={eventDates}
                        maxDate={maxDate}
                        minDate={minDate}
                        todayButton={false}
                        ranged
                        disabledDates={disabledDates}
                        startingWeekDay={startingWeekDay}
                        localization={localization}
                        locale={locale}
                        onChange={handleClickDate}
                    />
                </CalendarBox>
            )}
        </Box>
    );
};

DatepickerRange.displayName = 'DatepickerRange';
DatepickerRange.defaultProps = {
    isCalendarOpen: false,
};

export default DatepickerRange;
