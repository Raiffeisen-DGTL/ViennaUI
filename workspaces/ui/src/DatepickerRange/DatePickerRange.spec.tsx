/* eslint-disable @typescript-eslint/no-misused-promises */
import React from 'react';
import { DatepickerRange } from './DatepickerRange';

test('DatepickerRange', async () => {
    const snap = snapshot.render(<DatepickerRange />);
    expect(snap).toMatchSnapshot();
});

test('DatepickerRange /w not complete startDate', async () => {
    const snap = snapshot.render(<DatepickerRange value='22.02' />);
    expect(snap).toMatchSnapshot();
});

test('DatepickerRange /w complete startDate', async () => {
    const snap = snapshot.render(<DatepickerRange value='22.02.1988' />);
    expect(snap).toMatchSnapshot();
});

test('DatepickerRange /w not complete endDate', async () => {
    const snap = snapshot.render(<DatepickerRange value='22.02.1988-22' />);
    expect(snap).toMatchSnapshot();
});

test('DatepickerRange /w not complete', async () => {
    const snap = snapshot.render(<DatepickerRange value='11.12.1900-22.02.1988' />);
    expect(snap).toMatchSnapshot();
});

test('DatepickerRange /w date not in range', async () => {
    const snap = await snapshot.render(
        <DatepickerRange
            value={'01.06.2000 - 14.06.2000'}
            minDate={new Date(2000, 5, 2, 0, 0, 0, 0)}
            maxDate={new Date(2000, 5, 12, 0, 0, 0, 0)}
        />
    );
    expect(snap).toMatchSnapshot();
});

test('DatepickerRange /w invalid', async () => {
    const snap = snapshot.render(<DatepickerRange value={undefined} invalid />);
    expect(snap).toMatchSnapshot();
});
