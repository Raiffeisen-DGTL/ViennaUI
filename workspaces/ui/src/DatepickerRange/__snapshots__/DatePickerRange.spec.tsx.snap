// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatepickerRange /w complete startDate 1`] = `
<div
  class="DatepickerRangestyles__Box-sc-1dd67xt-0 hsiRJA"
>
  <div
    class="DatepickerRangestyles__InputBox-sc-1dd67xt-1 jqEJOL"
  >
    <div
      class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
    >
      <label
        class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
      >
        <div
          class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
        >
          <div
            class="Inputstyles__Box-hc8ttv-1 bzkyqR"
          >
            <div
              class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
            >
              <div
                class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
              >
                22.02.1988
              </div>
            </div>
            <input
              aria-invalid="false"
              class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
              value="22.02.1988"
            />
          </div>
        </div>
        <div
          class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
        >
          <svg
            cursor="pointer"
            fill="currentColor"
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs />
            <path
              d="M10 3v2h4V3h2v2h4a1 1 0 011 1v14a1 1 0 01-1 1H4a1 1 0 01-1-1V6a1 1 0 011-1h4V3h2zm8.999 10h-14L5 19h14l-.001-6zM16 15a1 1 0 110 2 1 1 0 010-2zM8 7H5l-.001 4h14L19 7h-3v2h-2V7h-4v2H8V7z"
            />
          </svg>
        </div>
      </label>
    </div>
  </div>
</div>
`;

exports[`DatepickerRange /w date not in range 1`] = `
<div
  class="DatepickerRangestyles__Box-sc-1dd67xt-0 hsiRJA"
>
  <div
    class="DatepickerRangestyles__InputBox-sc-1dd67xt-1 jqEJOL"
  >
    <div
      class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
    >
      <label
        class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
      >
        <div
          class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
        >
          <div
            class="Inputstyles__Box-hc8ttv-1 bzkyqR"
          >
            <div
              class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
            >
              <div
                class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
              >
                01.06.2000 - 14.06.2000
              </div>
            </div>
            <input
              aria-invalid="false"
              class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
              value="01.06.2000 - 14.06.2000"
            />
          </div>
        </div>
        <div
          class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
        >
          <svg
            cursor="pointer"
            fill="currentColor"
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs />
            <path
              d="M10 3v2h4V3h2v2h4a1 1 0 011 1v14a1 1 0 01-1 1H4a1 1 0 01-1-1V6a1 1 0 011-1h4V3h2zm8.999 10h-14L5 19h14l-.001-6zM16 15a1 1 0 110 2 1 1 0 010-2zM8 7H5l-.001 4h14L19 7h-3v2h-2V7h-4v2H8V7z"
            />
          </svg>
        </div>
      </label>
    </div>
  </div>
</div>
`;

exports[`DatepickerRange /w invalid 1`] = `
<div
  class="DatepickerRangestyles__Box-sc-1dd67xt-0 hsiRJA"
>
  <div
    class="DatepickerRangestyles__InputBox-sc-1dd67xt-1 jqEJOL"
  >
    <div
      class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
    >
      <label
        class="Inputstyles__Wrapper-hc8ttv-6 iaA-dFS"
      >
        <div
          class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
        >
          <div
            class="Inputstyles__Box-hc8ttv-1 bzkyqR"
          >
            <div
              class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
            >
              <div
                class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
              >
                ДД.ММ.ГГГГ
              </div>
            </div>
            <input
              aria-invalid="true"
              class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
              value=""
            />
          </div>
        </div>
        <div
          class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
        >
          <svg
            cursor="pointer"
            fill="currentColor"
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs />
            <path
              d="M10 3v2h4V3h2v2h4a1 1 0 011 1v14a1 1 0 01-1 1H4a1 1 0 01-1-1V6a1 1 0 011-1h4V3h2zm8.999 10h-14L5 19h14l-.001-6zM16 15a1 1 0 110 2 1 1 0 010-2zM8 7H5l-.001 4h14L19 7h-3v2h-2V7h-4v2H8V7z"
            />
          </svg>
        </div>
      </label>
    </div>
  </div>
</div>
`;

exports[`DatepickerRange /w not complete 1`] = `
<div
  class="DatepickerRangestyles__Box-sc-1dd67xt-0 hsiRJA"
>
  <div
    class="DatepickerRangestyles__InputBox-sc-1dd67xt-1 jqEJOL"
  >
    <div
      class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
    >
      <label
        class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
      >
        <div
          class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
        >
          <div
            class="Inputstyles__Box-hc8ttv-1 bzkyqR"
          >
            <div
              class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
            >
              <div
                class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
              >
                11.12.1900 - 22.02.1988
              </div>
            </div>
            <input
              aria-invalid="false"
              class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
              value="11.12.1900 - 22.02.1988"
            />
          </div>
        </div>
        <div
          class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
        >
          <svg
            cursor="pointer"
            fill="currentColor"
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs />
            <path
              d="M10 3v2h4V3h2v2h4a1 1 0 011 1v14a1 1 0 01-1 1H4a1 1 0 01-1-1V6a1 1 0 011-1h4V3h2zm8.999 10h-14L5 19h14l-.001-6zM16 15a1 1 0 110 2 1 1 0 010-2zM8 7H5l-.001 4h14L19 7h-3v2h-2V7h-4v2H8V7z"
            />
          </svg>
        </div>
      </label>
    </div>
  </div>
</div>
`;

exports[`DatepickerRange /w not complete endDate 1`] = `
<div
  class="DatepickerRangestyles__Box-sc-1dd67xt-0 hsiRJA"
>
  <div
    class="DatepickerRangestyles__InputBox-sc-1dd67xt-1 jqEJOL"
  >
    <div
      class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
    >
      <label
        class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
      >
        <div
          class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
        >
          <div
            class="Inputstyles__Box-hc8ttv-1 bzkyqR"
          >
            <div
              class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
            >
              <div
                class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
              >
                22.02.1988 - 22.ММ.ГГГГ
              </div>
            </div>
            <input
              aria-invalid="false"
              class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
              value="22.02.1988 - 22"
            />
          </div>
        </div>
        <div
          class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
        >
          <svg
            cursor="pointer"
            fill="currentColor"
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs />
            <path
              d="M10 3v2h4V3h2v2h4a1 1 0 011 1v14a1 1 0 01-1 1H4a1 1 0 01-1-1V6a1 1 0 011-1h4V3h2zm8.999 10h-14L5 19h14l-.001-6zM16 15a1 1 0 110 2 1 1 0 010-2zM8 7H5l-.001 4h14L19 7h-3v2h-2V7h-4v2H8V7z"
            />
          </svg>
        </div>
      </label>
    </div>
  </div>
</div>
`;

exports[`DatepickerRange /w not complete startDate 1`] = `
<div
  class="DatepickerRangestyles__Box-sc-1dd67xt-0 hsiRJA"
>
  <div
    class="DatepickerRangestyles__InputBox-sc-1dd67xt-1 jqEJOL"
  >
    <div
      class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
    >
      <label
        class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
      >
        <div
          class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
        >
          <div
            class="Inputstyles__Box-hc8ttv-1 bzkyqR"
          >
            <div
              class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
            >
              <div
                class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
              >
                22.02.ГГГГ
              </div>
            </div>
            <input
              aria-invalid="false"
              class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
              value="22.02"
            />
          </div>
        </div>
        <div
          class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
        >
          <svg
            cursor="pointer"
            fill="currentColor"
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs />
            <path
              d="M10 3v2h4V3h2v2h4a1 1 0 011 1v14a1 1 0 01-1 1H4a1 1 0 01-1-1V6a1 1 0 011-1h4V3h2zm8.999 10h-14L5 19h14l-.001-6zM16 15a1 1 0 110 2 1 1 0 010-2zM8 7H5l-.001 4h14L19 7h-3v2h-2V7h-4v2H8V7z"
            />
          </svg>
        </div>
      </label>
    </div>
  </div>
</div>
`;

exports[`DatepickerRange 1`] = `
<div
  class="DatepickerRangestyles__Box-sc-1dd67xt-0 hsiRJA"
>
  <div
    class="DatepickerRangestyles__InputBox-sc-1dd67xt-1 jqEJOL"
  >
    <div
      class="InputDateRangestyles__Box-z67vxn-0 dsxpIC"
    >
      <label
        class="Inputstyles__Wrapper-hc8ttv-6 daAZxl"
      >
        <div
          class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
        >
          <div
            class="Inputstyles__Box-hc8ttv-1 bzkyqR"
          >
            <div
              class="Inputstyles__PlaceholderWrapper-hc8ttv-7 frQCmf"
            >
              <div
                class="Inputstyles__Placeholder-hc8ttv-0 dDQzQZ"
              >
                ДД.ММ.ГГГГ
              </div>
            </div>
            <input
              aria-invalid="false"
              class="Inputstyles__StyledInput-hc8ttv-2 bUBKOu"
              value=""
            />
          </div>
        </div>
        <div
          class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
        >
          <svg
            cursor="pointer"
            fill="currentColor"
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs />
            <path
              d="M10 3v2h4V3h2v2h4a1 1 0 011 1v14a1 1 0 01-1 1H4a1 1 0 01-1-1V6a1 1 0 011-1h4V3h2zm8.999 10h-14L5 19h14l-.001-6zM16 15a1 1 0 110 2 1 1 0 010-2zM8 7H5l-.001 4h14L19 7h-3v2h-2V7h-4v2H8V7z"
            />
          </svg>
        </div>
      </label>
    </div>
  </div>
</div>
`;
